src/libraries/Common/src/Internal/Cryptography/AsymmetricAlgorithmHelpers.Hash.cs

```diff
@@ -21,10 +23,41 @@ public static byte[] HashData(byte[] data, int offset, int count, HashAlgorithmN
             Debug.Assert(offset >= 0 && offset <= data.Length - count);
             Debug.Assert(!string.IsNullOrEmpty(hashAlgorithm.Name));
 
+#if NET5_0_OR_GREATER
+            ReadOnlySpan<byte> source = data.AsSpan(offset, count);
+
+            if (hashAlgorithm == HashAlgorithmName.MD5)
+            {
+                return MD5.HashData(source);
+            }
+
+            if (hashAlgorithm == HashAlgorithmName.SHA1)
+            {
+                return SHA1.HashData(source);
+            }
+
+            if (hashAlgorithm == HashAlgorithmName.SHA256)
+            {
+                return SHA256.HashData(source);
+            }
+
+            if (hashAlgorithm == HashAlgorithmName.SHA384)
+            {
+                return SHA384.HashData(source);
+            }
+
+            if (hashAlgorithm == HashAlgorithmName.SHA512)
+            {
+                return SHA512.HashData(source);
+            }
```

Comment:
Minor, but should we reorder this based on most likely to be used first (I'm not sure what that order is, though :smile:)?  It could also be condensed if desired, e.g.
```C#
ReadOnlySpan<byte> source = data.AsSpan(offset, count);
return
    hashAlgorithm == HashAlgorithmName.SHA256 ? SHA256.HashData(source) :
    hashAlgorithm == HashAlgorithmName.SHA384 ? SHA256.HashData(source) :
    hashAlgorithm == HashAlgorithmName.SHA512 ? SHA512.HashData(source) :
    hashAlgorithm == HashAlgorithmName.SHA1 ? SHA1.HashData(source) :
    hashAlgorithm == HashAlgorithmName.MD5 ? MD5.HashData(source) :
    throw new CryptographicException(SR.Cryptography_UnknownHashAlgorithm, hashAlgorithm.Name);
```

Reply:
> should we reorder this based on most likely to be used first

I don't see why not. SHA256 / SHA1 is likely going to dominate most of the usage, so it makes sense to put them up first.

Reply:
What, no pattern switch statement? :)

(Kidding.)